internal/resolver/dns/dns_resolver_test.go

```diff
@@ -1443,7 +1443,7 @@ func TestCustomAuthority(t *testing.T) {
 		target := resolver.Target{
 			Endpoint:  "foo.bar.com",
 			Authority: a.authority,
-			URL:       url.URL{Host: a.authority},
+			URL:       url.URL{Host: a.authority, Path: "foo.bar.com"},
```

Comment:
Should this be replaced as well?

`URL: *testutils.MustParseURL(fmt.Sprintf("scheme://%s/foo.bar.com", a.authority))`

Reply:
Done - nice catch! 

I also updated `MustParseURL` to set the Host field to support `TestCustomAuthority`.

Reply:
Since there appears to be a hit to performance when using `fmt.Sprintf`, I opted to keep the string concatenation elsewhere in the file [like on line 737](https://github.com/grpc/grpc-go/pull/5852/files#diff-69918f87cc101780426a54cfb8c6b940ef83d827b8a10d1a3f8918236876fe27R737). Let me know if you'd like me to replace these lines with `fmt.Sprintf` to keep code consistent.

Thoughts? If I'm overthinking, please call me out. :)

Reply:
I wouldn't worry about performance too much here. These are tests anyways :)

Reply:
Ok - in that case I went ahead and refactored the remaining lines to also use `fmt.Sprintf` to keep it consistent. :)